engine/src/vault/view.rs

```diff
@@ -183,6 +196,82 @@ impl<P: BoxProvider> DbView<P> {
     pub fn clear(&mut self) {
         self.vaults.clear();
     }
+
+    /// List the ids of all vaults.
+    pub fn list_vaults(&self) -> Vec<VaultId> {
+        self.vaults.keys().cloned().collect()
+    }
+
+    /// List the ids of all records in the vault.
+    pub fn list_records(&self, vid: &VaultId) -> Vec<RecordId> {
+        self.vaults
+            .get(vid)
+            .map(|v| v.entries.keys().map(|&c| c.into()).collect())
+            .unwrap_or_default()
+    }
+
+    /// List [`RecordId`] and [`BlobId`] of all entries in the vault.
+    pub fn list_records_with_blob_id(
+        &self,
+        key: &Key<P>,
+        vid: VaultId,
+    ) -> Result<Vec<(RecordId, BlobId)>, VaultError<P::Error>> {
+        self.vaults
+            .get(&vid)
+            .ok_or(VaultError::VaultNotFound(vid))
+            .and_then(|v| v.list_entries(key).map_err(|e| e.into()))
+    }
+
+    /// Clone the all records from all vaults without removing them.
+    pub fn export_all(&self) -> HashMap<VaultId, Vec<(RecordId, Record)>> {
+        self.vaults
+            .iter()
+            .map(|(vid, vault)| {
+                let entries = vault.entries.iter().map(|(&id, r)| (id.into(), r.clone())).collect();
+                (*vid, entries)
+            })
+            .collect()
+    }
+
+    /// Clone the specified records from the vault without removing them.
+    ///
+    /// **Note:** before importing these records to a new vault with [`DbView::import_records`], [`Record::update_meta`]
+    /// has to be called on each [`Record`] to re-encrypt it with the target vault's encryption key.
```

Comment:
What happens if you export/import a record with the 'wrong' encryption key?  Is it just unreadable or is there an error?

Reply:
It will fail once you then try to read/ use the record, because the vault-key won't match. I originally implemented it so that the user has to provide the old encryption key and that the records are then re-encrypted with the new key on import. The problem is that on the higher layer on sync in #317with the mapper  it currently allows to re-arrange records to different locations in a very flexible way. With that, it could be that the list of records come from multiple different vaults and/ or clients. In that case we don't have a single "old" encryption key, but a map of keys, depending what the "old" location of the record was. Therefore I decided to move that logic into the sync-implementation. Will think about this a bit more and maybe change it again once I refactor the client-layer part of the sync feature.